File: OPENAPI-REST-API/openapi-client/php-dt/src/App/DTO/Website.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\PhpAttribute as DTA;

/**
 * Full detail view of a webhosting service including billing, service configuration, and panel access links.
 */
class Website
{
    #[DTA\Data(field: "serviceInfo", nullable: true)]
    #[DTA\Strategy("Object", ["type" => \App\DTO\WebsiteServiceInfo::class])]
    #[DTA\Validator("TypeCompliant", ["type" => \App\DTO\WebsiteServiceInfo::class])]
    public \App\DTO\WebsiteServiceInfo|null $service_info = null;

    #[DTA\Data(field: "client_links", nullable: true)]
    #[DTA\Strategy("Object", ["type" => \App\DTO\Collection133::class])]
    #[DTA\Validator("TypeCompliant", ["type" => \App\DTO\Collection133::class])]
    public \App\DTO\Collection133|null $client_links = null;

    #[DTA\Data(field: "billingDetails", nullable: true)]
    #[DTA\Strategy("Object", ["type" => \App\DTO\WebsiteBillingDetails::class])]
    #[DTA\Validator("TypeCompliant", ["type" => \App\DTO\WebsiteBillingDetails::class])]
    public \App\DTO\WebsiteBillingDetails|null $billing_details = null;

    /**
     * Customer&#39;s currency
     */
    #[DTA\Data(field: "custCurrency", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $cust_currency = null;

    /**
     * Customer currency symbol
     */
    #[DTA\Data(field: "custCurrencySymbol", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $cust_currency_symbol = null;

    #[DTA\Data(field: "serviceMaster", nullable: true)]
    #[DTA\Strategy("Object", ["type" => \App\DTO\WebsiteServiceMaster::class])]
    #[DTA\Validator("TypeCompliant", ["type" => \App\DTO\WebsiteServiceMaster::class])]
    public \App\DTO\WebsiteServiceMaster|null $service_master = null;

    /**
     * Package information
     */
    #[DTA\Data(field: "package", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $package = null;

    #[DTA\Data(field: "serviceExtra", nullable: true)]
    #[DTA\Strategy("Object", ["type" => \App\DTO\Collection134::class])]
    #[DTA\Validator("TypeCompliant", ["type" => \App\DTO\Collection134::class])]
    public \App\DTO\Collection134|null $service_extra = null;

    #[DTA\Data(field: "extraInfoTables", nullable: true)]
    #[DTA\Strategy("Object", ["type" => \App\DTO\WebsiteExtraInfoTables::class])]
    #[DTA\Validator("TypeCompliant", ["type" => \App\DTO\WebsiteExtraInfoTables::class])]
    public \App\DTO\WebsiteExtraInfoTables|null $extra_info_tables = null;

}

File: OPENAPI-REST-API/swagger-client/micronaut/src/main/java/io/swagger/model/Website.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.model.WebsiteBillingDetails;
import io.swagger.model.WebsiteClientLink;
import io.swagger.model.WebsiteExtraInfoTables;
import io.swagger.model.WebsiteServiceExtra;
import io.swagger.model.WebsiteServiceInfo;
import io.swagger.model.WebsiteServiceMaster;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.validation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Full detail view of a webhosting service including billing, service configuration, and panel access links.
 */
@Schema(description = "Full detail view of a webhosting service including billing, service configuration, and panel access links.")
@Validated
@Introspected

public class Website   {
  @JsonProperty("serviceInfo")
  private WebsiteServiceInfo serviceInfo = null;

  @JsonProperty("client_links")
  @Valid
  private List<WebsiteClientLink> clientLinks = null;

  @JsonProperty("billingDetails")
  private WebsiteBillingDetails billingDetails = null;

  @JsonProperty("custCurrency")
  private String custCurrency = null;

  @JsonProperty("custCurrencySymbol")
  private String custCurrencySymbol = null;

  @JsonProperty("serviceMaster")
  private WebsiteServiceMaster serviceMaster = null;

  @JsonProperty("package")
  private String _package = null;

  @JsonProperty("serviceExtra")
  @Valid
  private List<WebsiteServiceExtra> serviceExtra = null;

  @JsonProperty("extraInfoTables")
  private WebsiteExtraInfoTables extraInfoTables = null;

  public Website serviceInfo(WebsiteServiceInfo serviceInfo) {
    this.serviceInfo = serviceInfo;
    return this;
  }

  /**
   * Get serviceInfo
   * @return serviceInfo
  **/
  @Schema(description = "")
  @NotNull

  @Valid
  public WebsiteServiceInfo getServiceInfo() {
    return serviceInfo;
  }

  public void setServiceInfo(WebsiteServiceInfo serviceInfo) {
    this.serviceInfo = serviceInfo;
  }

  public Website clientLinks(List<WebsiteClientLink> clientLinks) {
    this.clientLinks = clientLinks;
    return this;
  }

  public Website addClientLinksItem(WebsiteClientLink clientLinksItem) {
    if (this.clientLinks == null) {
      this.clientLinks = new ArrayList<WebsiteClientLink>();
    }
    this.clientLinks.add(clientLinksItem);
    return this;
  }

  /**
   * Get clientLinks
   * @return clientLinks
  **/
  @Schema(description = "")
  @NotNull
  @Valid
  public List<WebsiteClientLink> getClientLinks() {
    return clientLinks;
  }

  public void setClientLinks(List<WebsiteClientLink> clientLinks) {
    this.clientLinks = clientLinks;
  }

  public Website billingDetails(WebsiteBillingDetails billingDetails) {
    this.billingDetails = billingDetails;
    return this;
  }

  /**
   * Get billingDetails
   * @return billingDetails
  **/
  @Schema(description = "")
  @NotNull

  @Valid
  public WebsiteBillingDetails getBillingDetails() {
    return billingDetails;
  }

  public void setBillingDetails(WebsiteBillingDetails billingDetails) {
    this.billingDetails = billingDetails;
  }

  public Website custCurrency(String custCurrency) {
    this.custCurrency = custCurrency;
    return this;
  }

  /**
   * Customer's currency
   * @return custCurrency
  **/
  @Schema(example = "USD", description = "Customer's currency")
  @NotNull

  public String getCustCurrency() {
    return custCurrency;
  }

  public void setCustCurrency(String custCurrency) {
    this.custCurrency = custCurrency;
  }

  public Website custCurrencySymbol(String custCurrencySymbol) {
    this.custCurrencySymbol = custCurrencySymbol;
    return this;
  }

  /**
   * Customer currency symbol
   * @return custCurrencySymbol
  **/
  @Schema(example = "$", description = "Customer currency symbol")
  @NotNull

  public String getCustCurrencySymbol() {
    return custCurrencySymbol;
  }

  public void setCustCurrencySymbol(String custCurrencySymbol) {
    this.custCurrencySymbol = custCurrencySymbol;
  }

  public Website serviceMaster(WebsiteServiceMaster serviceMaster) {
    this.serviceMaster = serviceMaster;
    return this;
  }

  /**
   * Get serviceMaster
   * @return serviceMaster
  **/
  @Schema(description = "")
  @NotNull

  @Valid
  public WebsiteServiceMaster getServiceMaster() {
    return serviceMaster;
  }

  public void setServiceMaster(WebsiteServiceMaster serviceMaster) {
    this.serviceMaster = serviceMaster;
  }

  public Website _package(String _package) {
    this._package = _package;
    return this;
  }

  /**
   * Package information
   * @return _package
  **/
  @Schema(example = "Web Hosting Direct Admin", description = "Package information")
  @NotNull

  public String getPackage() {
    return _package;
  }

  public void setPackage(String _package) {
    this._package = _package;
  }

  public Website serviceExtra(List<WebsiteServiceExtra> serviceExtra) {
    this.serviceExtra = serviceExtra;
    return this;
  }

  public Website addServiceExtraItem(WebsiteServiceExtra serviceExtraItem) {
    if (this.serviceExtra == null) {
      this.serviceExtra = new ArrayList<WebsiteServiceExtra>();
    }
    this.serviceExtra.add(serviceExtraItem);
    return this;
  }

  /**
   * Get serviceExtra
   * @return serviceExtra
  **/
  @Schema(description = "")
  @NotNull
  @Valid
  public List<WebsiteServiceExtra> getServiceExtra() {
    return serviceExtra;
  }

  public void setServiceExtra(List<WebsiteServiceExtra> serviceExtra) {
    this.serviceExtra = serviceExtra;
  }

  public Website extraInfoTables(WebsiteExtraInfoTables extraInfoTables) {
    this.extraInfoTables = extraInfoTables;
    return this;
  }

  /**
   * Get extraInfoTables
   * @return extraInfoTables
  **/
  @Schema(description = "")
  @NotNull

  @Valid
  public WebsiteExtraInfoTables getExtraInfoTables() {
    return extraInfoTables;
  }

  public void setExtraInfoTables(WebsiteExtraInfoTables extraInfoTables) {
    this.extraInfoTables = extraInfoTables;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Website website = (Website) o;
    return Objects.equals(this.serviceInfo, website.serviceInfo) &&
        Objects.equals(this.clientLinks, website.clientLinks) &&
        Objects.equals(this.billingDetails, website.billingDetails) &&
        Objects.equals(this.custCurrency, website.custCurrency) &&
        Objects.equals(this.custCurrencySymbol, website.custCurrencySymbol) &&
        Objects.equals(this.serviceMaster, website.serviceMaster) &&
        Objects.equals(this._package, website._package) &&
        Objects.equals(this.serviceExtra, website.serviceExtra) &&
        Objects.equals(this.extraInfoTables, website.extraInfoTables);
  }

  @Override
  public int hashCode() {
    return Objects.hash(serviceInfo, clientLinks, billingDetails, custCurrency, custCurrencySymbol, serviceMaster, _package, serviceExtra, extraInfoTables);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Website {\n");
    
    sb.append("    serviceInfo: ").append(toIndentedString(serviceInfo)).append("\n");
    sb.append("    clientLinks: ").append(toIndentedString(clientLinks)).append("\n");
    sb.append("    billingDetails: ").append(toIndentedString(billingDetails)).append("\n");
    sb.append("    custCurrency: ").append(toIndentedString(custCurrency)).append("\n");
    sb.append("    custCurrencySymbol: ").append(toIndentedString(custCurrencySymbol)).append("\n");
    sb.append("    serviceMaster: ").append(toIndentedString(serviceMaster)).append("\n");
    sb.append("    _package: ").append(toIndentedString(_package)).append("\n");
    sb.append("    serviceExtra: ").append(toIndentedString(serviceExtra)).append("\n");
    sb.append("    extraInfoTables: ").append(toIndentedString(extraInfoTables)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
